src/AutoMapper/MappingEngine.cs

```diff
@@ -212,28 +190,22 @@ public TDestination DynamicMap<TDestination>(object source)
 
         public object DynamicMap(object source, Type sourceType, Type destinationType)
         {
+            ConfigurationProvider.CreateMissingTypeMaps = true;

```

Comment:
Could you make a cached DynamicMappingProfile, with type map condition of true, instead and map using that?

It wouldn't share any prefix and postfixes of the default mapping, but I'm not sure what is to be expected when dynamic mapping.
I guess you could clone the default's properties and clear out TypeMap conventions and add `(sourceType, destType) => true` to achieve the default's behavior but override the confidence.  Or make a wrapper around the default that did everything except TypeMap conventions and override that.